gufe/chemicalsystem.py

```diff
@@ -72,6 +72,44 @@ def components(self) -> dict[str, Component]:
         """
         return dict(self._components)
 
+    def component_diff(self, other) -> dict[str, tuple[Component | None, Component | None]]:
+        """Compare the Components of this ChemicalSystem with the Components of another ChemicalSystem.
+
+        Parameters
+        ----------
+        other : ChemicalSystem
+            The ChemicalSystem to compare to.
+
+        Returns
+        -------
+        dict[str, tuple[Component | None, Component | None]]
+            A dictionary whose keys are the component labels present in either of the two ChemicalSystem objects.
+            The values are tuples containing the different Component objects from this and the other ChemicalSystem, respectively.
+            Only components that differ between the two ChemicalSystem objects will appear in the dictionary.
+
+        Raises
+        ------
+        TypeError
+            If `other` is not an instance of the same class as `self`.
+        """
+
+        # TODO: is this too restrictive?
+        if self.__class__ is not other.__class__:
```

Comment:
Can you change this to 
```suggestion
        if not isinstance(other, ChemicalSystem):
```

This would allow for subclasseses of `ChemicalSystem` to use this method.